### PR TITLE
Feature/bgm 673 update documentations on code repo

### PR DESCRIPTION
**Pull request by tracyarciaga**
## Description of Changes

Add documentation for diagrams and template for troubleshooting guide.
 
[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-36?atlOrigin=eyJpIjoiMTk1NTM4NGZiMzE4NGE1OGIwZWQzZTQyODUzN2Q3MjUiLCJwIjoiaiJ9)

### Developer Checklist:

* [ ] I have followed the guidelines in our Contributing document
* [ ] This PR has a corresponding JIRA ticket
* [ ] My branch conforms with our naming convention i.e. `feature/BGM-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] Code style confirms to eslint/flow rules
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

**Review by ernestpascual — COMMENTED** [review]
Same q from https://github.com/HaloDAO/emp-tools/pull/6#pullrequestreview-621153049!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!-- You can erase any parts of this template not applicable to your Pull Request. -->
+## Description of Changes
+<!-- Enter description of changes here. -->
+
+[Link to Jira Ticket](https://soowhq.atlassian.net/browse/SOOW-)
```
Comment (schystz) [inline .github/pull_request_template.md]: @tracyarciaga can we replace the link with `https://halodao.atlassian.net/browse/HDF-` so its easier to simply append the issue number at the end?

Comment (ctverceles) [reply .github/pull_request_template.md:5]: +1

Comment (chris247474) [reply .github/pull_request_template.md]: +1

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+* [ ] I have followed the guidelines in our Contributing document
+* [ ] This PR has a corresponding JIRA ticket
+* [ ] My branch conforms with our naming convention i.e. `feature/BGM-XXX-description`
+* [ ] All files have appropriate file headers and documentation
+* [ ] I have written new tests for your core changes, as applicable
+* [ ] I have successfully ran tests locally
+
+### Reviewers Checklist:
+* [ ] Code is readable and understandable; any unclear parts have explanations 
+* [ ] Code style confirms to eslint/flow rules
```
Comment (chris247474) [inline .github/pull_request_template.md]: Are we using eslint/flow now? Or maybe this is just leftover from the template. Can remove @tracyarciaga ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+* [Code quality](#code-quality) 
+* [Prettier](#Prettier) 
+* [Node version locking](#node-version-locking)
+
+
+## Required Software
+
+You will need the following things installed in order to run the app:
+
+- Nodejs `v8.16.2`. We **strongly** recommend you install via [NVM](https://github.com/creationix/nvm) to avoid incompatibility issues between different node projects.
```
Comment (chris247474) [inline CONTRIBUTING.md]: shall we change this to 14.15 @tracyarciaga @schystz @ernestpascual ? what are we all using for this repo?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+```
+   brew install nvm
+   nvm install v8.16.2
```
Comment (chris247474) [inline CONTRIBUTING.md]: same node version comment here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+   nvm use v8.16.2
```
Comment (chris247474) [inline CONTRIBUTING.md]: same node version comment here

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+- Check that the correct node (8.16.2) and npm (6.4.1) versions are installed
```
Comment (chris247474) [inline CONTRIBUTING.md]: same node version comment here

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    > Commits should be as fine-grained and focused as possible, it is not the place to be verbose. [read more...](https://medium.com/@preslavrachev/what-s-with-the-50-72-rule-8a906f61f09c)
+
+ * Capitalize the subject line.
+ * Do not end the subject line with a period.
+ * Use [imperative mood](https://en.wikipedia.org/wiki/Imperative_mood) in the subject line.
+
+## Coding Style Guidelines
+Code consistency, readability and maintainability are important to us. We strictly require you to check and follow the specific guidelines listed below. Clean code matters!
+#### Specific
+- [Solidity](https://docs.soliditylang.org/en/v0.8.2/style-guide.html)
```
Comment (chris247474) [inline CONTRIBUTING.md]: latest version here! https://docs.soliditylang.org/en/v0.8.3/style-guide.html

File: docs/diagrams.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+### Code Sequence Diagrams
+You can view the diagrams in our wiki page.
+
+ * [Stake BPT Earn HALO, Unstake BPT Stop Earning HALO](https://halodao.atlassian.net/wiki/spaces/HALODAO/pages/721251/Diagrams#Halo-Rewards%3A-Stake-BPT-Earn-HALO%2C-Unstake-BPT-Stop-Earning-HALO)
+ * [Deposit HALO, earn xHALO](https://halodao.atlassian.net/wiki/spaces/HALODAO/pages/721251/Diagrams#Halo-Chest%3A-Deposit-HALO%2C-earn-xHALO)
+ * [Deposit xHALO, withdraw HALO](https://halodao.atlassian.net/wiki/spaces/HALODAO/pages/721251/Diagrams#Halo-Chest%3A-Deposit-xHALO%2C-withdraw-HALO)
```
Comment (chris247474) [inline docs/diagrams.md:6]: great diagrams @tracyarciaga ! thanks for this initiative to make things clearer for everyone :)

**Review by chris247474 — CHANGES_REQUESTED** [review]
Very minor notes just to update some things and clarify others! Thanks for getting the docs started @tracyarciaga !

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+<!-- Enter description of changes here. -->
+
+[Link to Jira Ticket](https://soowhq.atlassian.net/browse/SOOW-)
+
+
+### Developer Checklist:
+
+* [ ] I have followed the guidelines in our Contributing document
+* [ ] This PR has a corresponding JIRA ticket
+* [ ] My branch conforms with our naming convention i.e. `feature/BGM-XXX-description`
```
Comment (chris247474) [inline .github/pull_request_template.md]: `feature/HDF-XXX-description`

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+#### React
+
+
+## Recommended Developer Tools & Configuration
+
+### Code quality
+
+#### Prettier
+
+You may follow base configurations for Prettier here https://github.com/prettier/prettier-vscode.
```
Comment (ernestpascual) [inline CONTRIBUTING.md:199]: @schystz  do we make our prettier config as the standard?

Comment (tracyarciaga) [reply CONTRIBUTING.md:199]: @ernestpascual I think we should

**Pull request merged**